### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by DocKDE** [dzfrias/projectable#24]
Not exactly a bug but in order to be actually useful (at least to me) the app would need to perform a bit better. 

When I fired it up in the folder containing all my dev projects (which is quite large) the performance was quite slow. It took ~ 2 seconds to start up and navigating the folders had a significant delay. I know this is not very specific but maybe lazy loading of the contents of subdirectories or some such might be an idea? Just my 2 cents though.

**Comment by dzfrias** [comment]
Yeah, you're completely right. I have some ideas for helping the startup time. Currently, the main bottleneck in that area is the path sorting algorithm. If I can find a way to speed it up (or just remove it), startup times should improve quite a bit. I'll also look into lazy loading if that doesn't work!

For navigation, the main problem is when drawing the file listing. Right now, projectable does a loop through each visible item so it can be drawn (every time the application needs to be redrawn). It's probably pretty costly, so I'll look into perhaps caching that?

**Comment by DocKDE** [comment]
This kind of stuff is probably a solved problem somewhere. There's a bunch of terminal-based file managers and the like and I suppose these have to deal with this as well. However, this would mean to rummage around in the source of all kinds of projects :D
